supersetapiclient/client.py

```diff
@@ -207,12 +208,21 @@ def _sql_endpoint(self) -> str:
 
     def csrf_token(self, session) -> str:
         # Get CSRF Token
-        csrf_response = session.get(
```

Comment:
This change can be simplified to just:

```python
+ if self.base_url.startswith('http:'):
+     os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
  csrf_response = session.get(...
```

That said, I'm not sure this package should be fiddling with `OAUTHLIB_INSECURE_TRANSPORT` on its own. It's a potential security risk and a global setting for `oauthlib` that would affect all other usage of `oauthlib` anywhere in the same process. I think it's better to let users handle this on their own to ensure they know what they're doing:

```python
scheme = ...
hostname = ...
port = ...
if scheme == "http":
    os.environ["OAUTHLIB_INSECURE_TRANSPORT"] = "1"

client = SupersetClient (host=f"{scheme}://{hostname}:{port}", ...)
```

Reply:
Thanks for your comment,
Yes you're right should be something that It handed at the user level more that in the library itself I'll remove these modifications from the MR and leave only the export bit